### PR TITLE
Use SqlDataTypeTest in ClickHouse mapping test

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #10363

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java [hunk 1]
```diff
@@ -36,24 +38,47 @@
...
+
+    // minutes offset change since 1970-01-01, no DST
+    private final ZoneId kathmandu = ZoneId.of("Asia/Kathmandu");
+
     private TestingClickHouseServer clickhouseServer;
 
+    @BeforeClass
+    public void setUp()
+    {
+        checkState(jvmZone.getId().equals("America/Bahia_Banderas"), "This test assumes certain JVM time zone");
```
Comment (grantatspothero) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java:69]: What is setting this JVM time zone?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java [hunk 2]
```diff
@@ -278,37 +316,35 @@ public void testTrinoVarbinary()
...
-            testCases.execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.test_date"));
-            testCases.execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"));
-        }
+    @DataProvider
+    public Object[][] testTimestampDataProvider()
+    {
+        return new Object[][] {
+                {UTC},
+                {jvmZone},
+                // using two non-JVM zones so that we don't need to worry what Oracle system zone is
```
Comment (grantatspothero) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java:343]: I think you copy pasted this comment from oracle data type tests.

**Review by grantatspothero — COMMENTED** [review]
Minor comment suggestions but the migration looks good.

Hopefully we can try out the new clickhouse jdbc driver when it is released soon: https://github.com/ClickHouse/clickhouse-jdbc/pull/770/files

**Review by Praveen2112 — APPROVED** [review]

**Pull request merged**